digitalaidseattle/seattle-humane-app
# Feature/automatic-db-type-verification

## What type of PR is this? (check all applicable)


## Description

This change adds a Github workflow to verify that the types are aligned between the Supabase configuration and the types in TypeScript. This workflow will execute whenever someone creates a pull request.

The `generate-types.yml` file is important here, the changes to the `database.types.ts` don't really matter, all that I did was up my local Supabase to version 1.150.0 and then run `supabase gen types typescript --local > supabase/database.types.ts`.

I followed the documentation [here](https://supabase.com/docs/guides/cli/github-action/generating-types), but needed to play with the setup a bit to get things working.

## Related Tickets & Documents

kinda 1.2.2.5, also 0.1.1.1

## QA Instructions, Screenshots, Recordings

I was able to test this by setting up Github workflows on a personal repo to see if how works. Then, I set it up on a branch in the seattle humane app and tried to get the scripts to behave as expected with the git check. You can probably find a bunch of failed workflows or maybe you got some emails.

You can test this by doing similar, if you change the action to trigger on push rather then on PR it's much easier.

## Added/updated tests?
_We are aiming for 100% statement coverage._

- [x] No, and this is why: This is basically a test as is, so it doesn't make sense to test it other than making sure it works

## Post-Merge tasks
I had deleted a migration file to get this to work: `20240309234741_all_tables_updated.sql`. We will need to add that behavior back.

## What gif best describes this PR or how it makes you feel?

![alt_text](https://media4.giphy.com/media/v1.Y2lkPTc5MGI3NjExYmk0cm92c2tkbGo2YTlxbnY3NDFocXdjMTdwMW90MHluZnA1em5rNiZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/xkmQfH1TB0dLW/giphy.gif)
